File: 30days/day10-binary-numbers/Solution.js
// PROBLEM: https://www.hackerrank.com/challenges/30-binary-numbers/problem

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}



function main() {
    const n = parseInt(readLine(), 10);
    const nbase2 = n.toString(2);
    const consecutive1s = nbase2.split('0').map(m => m.length);
    
    console.log(Math.max(...consecutive1s));
}

File: 30days/day10-binary-numbers/Solution.cs
// PROBLEM: https://www.hackerrank.com/challenges/30-binary-numbers/problem

using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {



    static void Main(string[] args) {
        int n = Convert.ToInt32(Console.ReadLine());
        string nbase2 = Convert.ToString(n, 2);
        IEnumerable<int> consecutive1s = nbase2.Split('0').Select(s => s.Length);
        
        Console.WriteLine(consecutive1s.Max());
    }
}
